GitFyle.Core.Api/Services/Foundations/Configurations/ConfigurationService.Exceptions.cs

```diff
@@ -0,0 +1,133 @@
+                        message: "Configuration already exists error occurred.",
+                        innerException: duplicateKeyException,
+                        data: duplicateKeyException.Data);
+
+                throw await CreateAndLogDependencyValidationExceptionAsync(alreadyExistsConfigurationException);
+            }
+            catch (DbUpdateException dbUpdateException)
+            {
+                var failedOperationConfigurationException =
+                    new FailedOperationConfigurationException(
+                        message: "Failed operation configuration error occurred, contact support.",
+                        innerException: dbUpdateException);
+
+                throw await CreateAndLogDependencyExceptionAsync(failedOperationConfigurationException);
+            }
+            catch (Exception serviceException)
+            {
+                var failedServiceConfigurationException =
+                    new FailedServiceConfigurationException(
+                        message: "Failed service configuration error occurred, contact support.",
+                        innerException: serviceException);
+
+                throw await CreateAndLogServiceExceptionAsync(failedServiceConfigurationException);
+            }
+        }
+
+        private async ValueTask<ConfigurationServiceException> CreateAndLogServiceExceptionAsync(Xeption exception)
+        {
+            var configurationServiceException =
+                new ConfigurationServiceException(
+                    message: "Service error occurred, contact support.",
+                    innerException: exception);
+
+            await this.loggingBroker.LogErrorAsync(configurationServiceException);
+
+            return configurationServiceException;
+        }
+
+        private async ValueTask<ConfigurationDependencyException> CreateAndLogDependencyExceptionAsync(Xeption exception)
+        {
+            var configurationDependencyException =
+                new ConfigurationDependencyException(
+                    message: "Configuration dependency error occurred, contact support.",
+                    innerException: exception);
+
+            await this.loggingBroker.LogErrorAsync(configurationDependencyException);
+
+            return configurationDependencyException;
+        }
+
+        private async ValueTask<ConfigurationDependencyValidationException> CreateAndLogDependencyValidationExceptionAsync(Xeption exception)
+        {
+            var configurationDependencyValidationException =
+                new ConfigurationDependencyValidationException(
+                    message: "Configuration dependency validation error occurred, fix errors and try again.",
+                    innerException: exception);
+
+            await this.loggingBroker.LogErrorAsync(configurationDependencyValidationException);
+
+            return configurationDependencyValidationException;
+        }
```

Comment:
Signature line over 120


```suggestion
        private async ValueTask<ConfigurationDependencyValidationException> CreateAndLogDependencyValidationExceptionAsync(
            Xeption exception)
        {
            var configurationDependencyValidationException =
                new ConfigurationDependencyValidationException(
                    message: "Configuration dependency validation error occurred, fix errors and try again.",
                    innerException: exception);

            await this.loggingBroker.LogErrorAsync(configurationDependencyValidationException);

            return configurationDependencyValidationException;
        }
```

Reply:
Done

Reply:
Moved Method name to next line.